### DISCUSSION TITLE
CloudFormation Analysis

### DISCUSSION THREAD

**Discussion by djfurman** [open-policy-agent/community#115] [OPA and Rego]
I'm really impressed with OPA and I want to apply it to CloudFormation ChangeSets. I see a tutorial on how to apply it to terraform plan and I'd like to follow the same example, but I've struggled with where the `import terraform.analysis` package code actually lives and where I could try to raise a PR to add this functionality for analyzing CloudFormation.

**Comment by srenatus** [comment]
>  I've struggled with where the import terraform.analysis package code actually lives

Those are a bunch of helper modules defined in the docs for terraform: https://www.openpolicyagent.org/docs/v0.37.1/terraform/#3-write-the-opa-policy-to-check-the-plan. Look for the code block starting with `package terraform.analysis`.

AFAIK they aren't defined anywhere else, and would be used by copying them into your local set of rego files.

If you want to publicise using OPA with CloudFormation, I suppose one thing you could do is adding a similar tutorial to the OPA docs; or publish it in your blog, or your company's blog, and add that to [the ecosystem page](https://www.openpolicyagent.org/docs/edge/ecosystem/).
